modules/libraries/aws/ota/src/aws_iot_ota_agent.c

```diff
@@ -533,15 +587,54 @@ OTA_State_t OTA_AgentInit( void * pvClient,
     /* The actual OTA queue control structure. Only created once. */
     static StaticQueue_t xStaticQueue;
 
-    /* Set the function to be called after an OTA job is complete or starting test mode. */
-    if( xFunc == NULL )
-    {
-        /* If the caller didn't supply a callback function, provide a default handler. */
-        xOTA_Agent.pxOTAJobCompleteCallback = prvDefaultOTACompleteCallback;
-    }
-    else
+    /* Check all the callbacks for null values and initialize the values in the ota agent context 
+       The OTA agent context is initialized with the prvPAL values. So, if null is passed in, don't
+       do anything and just use the defaults in the OTA structure
+    */
+
+    if ( xCallbacks != NULL )
     {
-        xOTA_Agent.pxOTAJobCompleteCallback = xFunc;
+        if( xCallbacks->xAbort != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xAbort = xCallbacks->xAbort;
+        }
+        if( xCallbacks->xActivateNewImage != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xActivateNewImage = xCallbacks->xActivateNewImage;
+        }
+        if( xCallbacks->xCloseFile != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xCloseFile = xCallbacks->xCloseFile;
+        }
+        if( xCallbacks->xCreateFileForRx != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xCreateFileForRx = xCallbacks->xCreateFileForRx;
+        }
+        if( xCallbacks->xGetPlatformImageState != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xGetPlatformImageState = xCallbacks->xGetPlatformImageState;
+        }
+        if( xCallbacks->xResetDevice != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xResetDevice = xCallbacks->xResetDevice;
+        }
+        if( xCallbacks->xSetPlatformImageState != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xSetPlatformImageState = xCallbacks->xSetPlatformImageState;
+        }
+        if( xCallbacks->xWriteBlock != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xWriteBlock = xCallbacks->xWriteBlock;
+        }
+        /* Set the function to be called after an OTA job is complete or starting test mode. */
+        if( xCallbacks->xCompleteCallback != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xCompleteCallback = xCallbacks->xCompleteCallback;
+        }
+        if( xCallbacks->xCustomJobCallback != NULL )
+        {
+            xOTA_Agent.xPALCallbacks.xCustomJobCallback = xCallbacks->xCustomJobCallback;
+        }
     }
```

Comment:
Take care of the else statement - if the callbacks are null, use the default callback

Reply:
> Take care of the else statement - if the callbacks are null, use the default callback

If it is NULL, the defaults are already filled into the OTA Agent context. Nothing needs to be done